files/scripts/syncd_init_common.sh

```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/env bash
+
+CMD_SYNCD=/usr/bin/syncd
+
+# dsserve: domain socket server for stdio
+CMD_DSSERVE=/usr/bin/dsserve
+CMD_DSSERVE_ARGS="$CMD_SYNCD --diag"
+
+ENABLE_SAITHRIFT=0
+
+PLATFORM_DIR=/usr/share/sonic/platform
+HWSKU_DIR=/usr/share/sonic/hwsku
+
+SONIC_ASIC_TYPE=$(sonic-cfggen -y /etc/sonic/sonic_version.yml -v asic_type)
+
+if [ -x $CMD_DSSERVE ]; then
+    CMD=$CMD_DSSERVE
+    CMD_ARGS=$CMD_DSSERVE_ARGS
+else
+    CMD=$CMD_SYNCD
+    CMD_ARGS=
+fi
+
+# Use temporary view between init and apply
+CMD_ARGS+=" -u"
+
+# Set synchronous mode if it is enabled in CONFIG_DB
+SYNC_MODE=$(sonic-cfggen -d -v DEVICE_METADATA.localhost.synchronous_mode)
+if [ "$SYNC_MODE" == "enable" ]; then
+    CMD_ARGS+=" -s"
+fi
```

Comment:
Only this part should be moved into buildimage repo, and all others belong more into original sairedis repo.

Reply:
I think probably a better way could be putting this part in syncd_start.sh and move it to buildimage repo. All other parts could remain unchanged.

Reply:
I will try if adding it in syncd_start.sh looks better.